extensions/lifecycle/conductor/service.js

```diff
@@ -1,44 +1,82 @@
 'use strict'; // eslint-disable-line
 
+const async = require('async');
 const werelogs = require('werelogs');
+const { errors } = require('arsenal');
+const { DEFAULT_LIVE_ROUTE, DEFAULT_READY_ROUTE } = require('arsenal').network.probe.ProbeServer;
+const { ZenkoMetrics } = require('arsenal').metrics;
 
 const LifecycleConductor = require('./LifecycleConductor');
-const { HealthProbeServer } = require('arsenal').network.probe;
-
 const config = require('../../../lib/Config');
+const { sendSuccess, sendError, startProbeServer } = require('../../../lib/util/probe');
+
 const zkConfig = config.zookeeper;
 const kafkaConfig = config.kafka;
 const lcConfig = config.extensions.lifecycle;
 const repConfig = config.extensions.replication;
 
-const lcConductor = new LifecycleConductor(
-    zkConfig, kafkaConfig, lcConfig, repConfig);
 
-werelogs.configure({ level: config.log.logLevel,
-                     dump: config.log.dumpLevel });
-const logger = new werelogs.Logger('Backbeat:Lifecycle:Conductor:service');
-const healthServer = new HealthProbeServer({
-    bindAddress: config.healthcheckServer.bindAddress,
-    port: config.healthcheckServer.port,
+const lcConductor = new LifecycleConductor(zkConfig, kafkaConfig, lcConfig, repConfig);
+
+werelogs.configure({
+    level: config.log.logLevel,
+    dump: config.log.dumpLevel,
 });
+const logger = new werelogs.Logger('Backbeat:Lifecycle:Conductor:service');
+
+/**
+ * Handle ProbeServer liveness check
+ *
+ * @param {http.HTTPServerResponse} res - HTTP Response to respond with
+ * @param {Logger} log - Logger
+ * @returns {string} response
+ */
+function handleLiveness(res, log) {
+    if (lcConductor.isReady()) {
+        sendSuccess(res, log);
+    } else {
+        log.error('lifecycle conductor is not ready!');
+        sendError(res, log, errors.ServiceUnavailable, 'unhealthy');
+    }
+}
 
-lcConductor.start(err => {
+/**
+ * Handle ProbeServer metrics
+ *
+ * @param {http.HTTPServerResponse} res - HTTP Response to respond with
+ * @param {Logger} log - Logger
+ * @returns {string} response
+ */
+function handleMetrics(res, log) {
```

Comment:
hmm on second through, we may want to skip this method for now as there are no prom metrics yet

Reply:
The existing feature does produce some metrics, so retaining it.

Reply:
oh i did not know that